### ISSUE TITLE
EasyBuild 4.4.0 crashes with `can't pickle` under some circumstances

### ISSUE THREAD

**Issue by damianam** [easybuilders/easybuild-framework#3727]
Example:

```
$ eb Julia-1.6.1-gomkl-2020.eb -r ../../ --include-toolchains=../../../Custom_Toolchains/*py,../../../Custom_Toolchains/compiler/*py --include-easyblocks=../../../Custom_EasyBlocks/*py -D
== Temporary log file in case of crash /tmp/eb-aepq9u78/easybuild-kulpj25r.log
Traceback (most recent call last):
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/main.py", line 548, in <module>
    main()
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/main.py", line 381, in main
    easyconfigs, generated_ecs = parse_easyconfigs(paths, validate=not options.inject_checksums)
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/framework/easyconfig/tools.py", line 395, in parse_easyconfigs
    easyconfigs.extend(process_easyconfig(ec_file, **kwargs))
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/framework/easyconfig/easyconfig.py", line 2018, in process_easyconfig
    ec = EasyConfig(spec, build_specs=build_specs, validate=validate, hidden=hidden)
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/framework/easyconfig/easyconfig.py", line 514, in __init__
    self.parse()
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/framework/easyconfig/easyconfig.py", line 698, in parse
    ec_vars = self.parser.get_config_dict()
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/framework/easyconfig/parser.py", line 222, in get_config_dict
    cfg = self._formatter.get_config_dict()
  File "/p/software/juwels/stages/Devel-2020/software/EasyBuild/4.4.0/lib/python3.6/site-packages/easybuild/framework/easyconfig/format/one.py", line 139, in get_config_dict
    cfg_copy[key] = copy.deepcopy(cfg[key])
  File "/usr/lib64/python3.6/copy.py", line 169, in deepcopy
    rv = reductor(4)
TypeError: can't pickle module objects
```

The environment is empty (no `$EASYBUILD_*` variables). Trying to narrow it down now.

**Comment by damianam** [comment]
So the EasyConfig affected have things like:

```import os as local_os
[...]
cuda_compute_capabilities = {
    'juwels': '7.0',
    'juwelsbooster': '8.0',
    'jurecadc': '8.0',
    'jusuf': '7.0',
    'hdfml': '7.0'
}[local_os.environ['SYSTEMNAME']]```

And that breaks the deep copy

**Comment by damianam** [comment]
https://github.com/easybuilders/easybuild-framework/pull/3729

**Comment by boegel** [comment]
fixed in #3729, so closing...
